samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -52,8 +52,11 @@
   private final StreamManager streamManager;
   private final SystemAdmins systemAdmins;
 
+  protected final StreamGraphImpl streamGraph;
```

Comment:
Not just for testing any more. This is used to directly instantiate TaskFactory. I will work on separating the TaskFactory/AppRunner related changes outside this PR as much as possible, to avoid confusion.